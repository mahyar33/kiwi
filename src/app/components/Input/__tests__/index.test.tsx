import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Input } from '../index';

const renderWithProps = (props: Parameters<typeof Input>[number]) =>
  render(<Input {...props} />);

describe('<Input />', () => {
  const props: Parameters<typeof Input>[number] = {
    type: 'text',
    placeholder: 'searchP',
  };
  it('should render an <input> tag', () => {
    const input = renderWithProps(props);
    expect(input.container.querySelector('input')).toBeInTheDocument();
  });
  it('should match snapshot', () => {
    const input = renderWithProps(props);
    expect(input.container.firstChild).toMatchSnapshot();
  });
  it('should have props', () => {
    const input = renderWithProps(props);
    expect(input.container.querySelector('input')).toHaveAttribute(
      'type',
      props.type,
    );
    expect(input.container.querySelector('input')).toHaveAttribute(
      'placeholder',
      props.placeholder,
    );
  });
  it('types inside input', () => {
    const input = renderWithProps(props);
    userEvent.type(input.container.querySelector('input')!, 'Hello, World!');
    expect(input.container.querySelector('input')!).toHaveValue(
      'Hello, World!',
    );
  });
  it('should call onChange', () => {
    props.onChange = jest.fn();
    const input = renderWithProps(props);
    userEvent.type(input.container.querySelector('input')!, 'Hello, World!');
    expect(props.onChange).toHaveBeenCalled();
  });
});
