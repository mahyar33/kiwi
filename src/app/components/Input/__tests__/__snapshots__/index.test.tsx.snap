// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot 1`] = `
.c0 {
  display: block;
  width: 100%;
  height: 2.5rem;
  padding: 6px 12px;
  line-height: 1.5;
  color: black;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ccc;
}

.c0:focus-visible {
  outline: unset;
}

<input
  class="c0"
  placeholder="searchP"
  type="text"
  value=""
/>
`;
